dev/com.ibm.ws.cdi.interfaces/src/com/ibm/wsspi/cdi/extension/WebSphereCDIExtensionMetaData.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2015 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.cdi.spi;
+
+import java.util.Collections;
+import java.util.Set;
+
+import javax.enterprise.inject.spi.Extension;
+
+/**
+ * This is a *marker* interface for CDI Runtime extensions. Liberty features that wish to extend CDI will need to
+ * register a service under this interface.
+ *
+ * To use this class you must implement one of the two methods. If you implement getBeanClasses() all classes returned by that
+ * method will be registered with CDI and may be used normally by application code. If you implement getExtensions() then
+ * all classes returned by that method will be treated as CDI extensions and run when creating the CDI container for each application.
+ *
+ * Classes returned from getExtensions() must implement javax.enterprise.inject.spi.Extension. They do *not* and should not be listed in a
+ * META-INF/services file. It is best practice to not put CDIExtensionMetaData and javax.enterprise.inject.spi.Extension on the same class
+ * as that would result in CDI and OSGI independently instantiating the class. Even though it is unlikely for the two instances to conflict, 
+ * it is best to keep the OSGI service and CDI extension separate. 
+ *
+ * The class that implements this interface should be annotated with @Component(service = WebSphereCDIExtensionMetaData.class). There are no
+ * other properties that can be included in the Component annotation.
+ *
+ * Here is a worked example of a complete WebSphereCDIExtensionMetaData implementation.
+ *
+ * @Component(service = WebSphereCDIExtensionMetaData.class)
+ * public class SPIMetaData implements WebSphereCDIExtensionMetaData {
+ *
+ *     @Override
+ *     public Set<Class<?>> getBeanClasses() {
+ *         Set<Class<?>> beans = new HashSet<Class<?>>();
+ *         //This will register a producer class and expose it's produced beans to applications
+ *         beans.add(ClassSPIRegisteredProducer.class);
+ *     }
+ * }
```

Comment:
Javadoc's rendering sucks, I think you need to do this here:

```suggestion
 * <pre>
 * @Component(service = WebSphereCDIExtensionMetaData.class)
 * public class SPIMetaData implements WebSphereCDIExtensionMetaData {
 *
 *     @Override
 *     public Set&lt;Class&lt;?&gt;&gt; getBeanClasses() {
 *         Set&lt;Class&lt;?&gt;&gt; beans = new HashSet&lt;Class&lt;?&gt;&gt;();
 *         //This will register a producer class and expose it's produced beans to applications
 *         beans.add(ClassSPIRegisteredProducer.class);
 *     }
 * }
 * </pre>
```

Need to check this renders correctly when generated as html.